tower-buffer/src/worker.rs

```diff
@@ -157,30 +165,43 @@ where
 
         loop {
             match try_ready!(self.poll_next_msg()) {
-                Some(msg) => {
+                Some((msg, first)) => {
+                    let _guard = msg.span.enter();
                     if let Some(ref failed) = self.failed {
+                        tracing::trace!("notifying about worker failure");
                         let _ = msg.tx.send(Err(failed.clone()));
                         continue;
                     }
 
                     // Wait for the service to be ready
+                    tracing::trace!(
+                        resumed = !first,
+                        message = "worker received request; waiting for service readiness"
+                    );
                     match self.service.poll_ready() {
                         Ok(Async::Ready(())) => {
+                            tracing::debug!("processing request");
                             let response = self.service.call(msg.request);
 
                             // Send the response future back to the sender.
                             //
                             // An error means the request had been canceled in-between
                             // our calls, the response future will just be dropped.
+                            tracing::trace!("returning response future");
                             let _ = msg.tx.send(Ok(response));
                         }
                         Ok(Async::NotReady) => {
+                            tracing::trace!("service not ready; delay");
```

Comment:
As I mentioned above, what if this was
```suggestion
                            tracing::trace!(service.ready = false, message = "delay");
```